### PR TITLE
Add org wild notifier to org layer

### PR DESCRIPTION
**Pull request by dalanicolai**
This PR replaces  #14396. See last comments there for motivation.

I've added instructions to use =org-journal-new-scheduled-entry=, assuming that PR numbers [335](https://github.com/bastibe/org-journal/pull/335), [337](https://github.com/bastibe/org-journal/pull/337) and [338](https://github.com/bastibe/org-journal/pull/338) get merged soon.

### CODE REVIEW

File: layers/+emacs/org/packages.el [hunk 1]
```diff
@@ -588,6 +590,12 @@ Headline^^            Visit entry^^               Filter^^                    Da
       (kbd "M-SPC") 'spacemacs/org-agenda-transient-state/body
       (kbd "s-M-SPC") 'spacemacs/org-agenda-transient-state/body)))
 
+(defun org/init-org-wild-notifier ()
+  (use-package org-wild-notifier
+    :init
```
Comment (smile13241324) [inline layers/+emacs/org/packages.el:595]: I think this package should be deferred.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
Looks good, apart from that the new package should be deferred.
Can you close your old PR if it is not longer needed?

File: layers/+emacs/org/packages.el (hunk 1 above)
Comment (dalanicolai) [reply layers/+emacs/org/packages.el:596]: Ah don't know why I left it out, but should be there indeed. I have pushed the update. Thanks again for checking and feedback!

**Comment by dalanicolai** [comment]
Closing this PR in favor of PR  #14418

**Review by smile13241324 — APPROVED** [review]
Looks good, will be merged soon.

**Comment by smile13241324** [comment]
@dalanicolai looks like there are some conflicts in your PR, can you have a look? Especially there is a merge which prevents me from cherry picking your commit. Can you squash/fix the conflicts on your side?

**Comment by dalanicolai** [comment]
Ah sorry. I am still no git expert. To create the `org-wild-notifier` fix, I branched from develop but then I merged from the previous `add-notify` PR (my previous PR to add `add-notify`, that I replaced with `org-wild-notifier` in this PR) so that I did not have to type everything new again. However, I guess I should just have had to use `git apply` (or use git -reset ^HEAD after the merge). Anyway, I am trying to learn and understand it (but I have only internet as my teacher).

Therewere some more git horrors when I tried to rebase, because somehow it did not leave my 'fix branch' intact. Anyway. I have applied some tricks now to fix it. I hope I have done things more or less correctly now.

Thanks again of course!

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Comment by dalanicolai** [comment]
The documentation part in this PR assumes that https://github.com/bastibe/org-journal/pull/339 got merged, which did not happen yet. But, although I expected it to get merged sooner, I still expect it to get merged very soon (probably before anybody will discover this documentation about those `org-journal` commands).

**Pull request closed without merging**